server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -835,7 +835,20 @@ public Iterator<Setting<?>> settings() {
                 return Boolean.FALSE.toString();
             }
             var indexMode = IndexSettings.MODE.get(settings);
-            return Boolean.toString(indexMode.useTimeSeriesDocValuesCodec());
+            return Boolean.toString(indexMode == IndexMode.TIME_SERIES && indexMode.useCustomDocValuesCodec());
+        },
+        Property.IndexScope,
+        Property.Final
+    );
+
+    public static final Setting<Boolean> USE_LOGS_DOC_VALUES_FORMAT_SETTING = Setting.boolSetting(
```

Comment:
Let's not use logs in the name here. I think we just want to control whether larger block size is used when time series codec is used. So something like this:

```
public static final Setting<Boolean> TIME_SERIES_DV_FORMAT_USE_LARGER_BLOCK_SIZE = Setting.boolSetting(
"index.time_series_doc_values_format_use_larger_block_size",
        settings -> {
            if (settings == null) {
                return Boolean.FALSE.toString();
            }
            var indexMode = IndexSettings.MODE.get(settings);
            return indexMode == IndexMode.TIME_SERIES;
        },
        Property.IndexScope,
        Property.Final
}
```

This is more of an escape hatch setting.